### ISSUE TITLE
Corner cases for User Wizard Flow

### ISSUE THREAD

**Issue by okaycj** [lookit/lookit-api#956]
# TL;DR
Here are some corner cases that were though of during developing the new User Wizard flow. 

# Narrative
QA'ing is hard and this is an attempt to make sure any corner cases with the new User Wizard flow are handled.  

# Acceptance Criteria
Corner cases that should probably be either talked about or handled:

- [x] What happens when there is Demographic data but no Child and user goes to study? User is prompted to create a child. 
- [x] When do we clear the study info from session?  After user attempts any study (Internal or external).
- [x] What if a user is directed to a study that doesn't exist?  User can't go to a study that doesn't exists.  
- [x] If you already have child and user clicks `save` button on Demographics view, they're redirected to study list.  Clicking `save` should probably always go back to the Demographics view.
